### DISCUSSION TITLE
Access-level analysis

### DISCUSSION THREAD

**Discussion by there-it-is** [peripheryapp/periphery#311] [Ideas]
Hello periphery, 

Your tool is great and I have found a lot of improvements in my current project with periphery.
Currently I'm searching for the ability to analyse the access-level modfiers of the methods but I didn't found any tool for this use case.
Maybe I have missed it in the periphery configuration but it would be great if this would work with periphery.

Like a message that for a specific method (and variable / class) the access modifier "private" would be enough.

Greetings
Thomas

**Comment by ileitch** [comment]
Hey Thomas,

Great to hear you're enjoying using Periphery!

I just released version 2.5.0 yesterday that can identify redundant `public` accessibility, however there's currently no analysis to detect redundant `internal` accessibility. It's a feature I plan to add soon though.
